flink-connectors/flink-connector-kafka-0.9/src/main/java/org/apache/flink/streaming/connectors/kafka/config/RateLimitingConfig.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.connectors.kafka.config;
+
+import org.apache.flink.streaming.api.operators.StreamingRuntimeContext;
+
+import java.util.Properties;
+
+/**
+ * A generic RateLimitingConfig that can be used to set ratelimiting properties across
+ * consumers.
+ */
+public class RateLimitingConfig {
```

Comment:
De-static-ify this class, please.  @glaksh100 and I talked offline about this.  Just noting here for visibility.  This class will basically become a Factory that will build a already configured RateLimiter.  Easy peasy.

Reply:
If you can configure that as consumer property then this would also remove the Guava dependency and allow others to plug in their own implementation.